class Solution {
    public boolean checkValidString(String s) {
      Stack<Integer> open =new Stack<>();
      Stack<Integer> star=new Stack<>();

      for(int i=0; i<s.length(); i++){
      char ch=s.charAt(i);
      if(ch=='*') star.push(i);

      else if(ch=='(') open.push(i);

      else{
        if(!open.isEmpty()) open.pop();

        else if(!star.isEmpty()) star.pop();

        else  return false;
      }
      }  

      while(!star.isEmpty() && !open.isEmpty()){
      if(star.peek()>open.peek()) {
        open.pop();
        star.pop();
      }
      else{
        return false;
      }
      }
 return open.isEmpty();   }
}